katana/native/file_manager/file_manager_utils/ftp_utils.py

```diff
@@ -0,0 +1,153 @@
+                    else:
+                        result.append(name + ': Is not a file or directory')
+            else:
+                result.append(destdir + " :Could Not Change to Destination Directory Successfully")
+                return result
+        except ftplib.all_errors as e:
+            result.append(str(e))
+        ftp.quit()
+    except ftplib.all_errors as e:
+        result.append(str(e))
+    return result
+
+
+def change_destination(ftp, destdir):
+    """
+    If the destiantion directory is different than the root in the target machine, then this function traverses into
+    the destination directory in the target machine.
+
+    :param ftp: FTP object
+    :param destdir: Destination directory from the root directory where the files are to be FTPed.
+    :return: Boolean.
+    """
+    try:
+        if destdir != '':
+            ftp.cwd(destdir)
+    except:
+        print('\nCould not change to destination directory')
+        return False
+    return True
+
+
+def transfer_file(ftp, path, name):
+    """
+    This function FTPs the file. Some files have to be sent as binary format and some as ascii format.
+
+    :param ftp: FTP object
+    :param path: Full path to file name
+    :param name: File name
+    :return: Status of the file transfer
+    """
+    fp = open(path, 'rb')
+    try:
+        ext = os.path.splitext(name)[1]
+        # check if the extension needs to be sent in binary mode. Default is ASCII mode.
+        if ext == '.docx' or ext == '.doc':
+            ftp.storbinary('STOR ' + name, fp)
+        else:
+            ftp.storlines('STOR ' + name, fp)
+        # storlines prints new line character at the end of each line in text file
+        # but it does not transfer word files.
+    except:
+        return name + ': Could not transfer the file.'
+    fp.close()
+    return name + ' :Success'
+
+
+def isDir(ftp, path, name, files_path, files_name, all_files_path):
+    """
+    If the selected item is a directory, then this is a recursive function that FTPs all the files selected within
+    the directory. If the item selected is a directory, then this function is called on that directory.
+
+    The Directory structure is recreated in the target machine.
+
+    This also checks if there are any broken tree files. These are files which are not associated to a parent but has
+    some ancestor selected.
+
+    :param ftp: FTP object
+    :param path: Full path of the directory
+    :param name: Name of the directory
+    :param files_path: Full path of the items selected within the directory
+    :param files_name: Names of the items selected within the directory
+    :param all_files_path: all the items in the directory
+    :return: Status or error
+    """
+    result = []
+    parent_dir = ftp.pwd()
+    if parent_dir == '/':
+        new_folder = parent_dir + name
+    else:
+        new_folder = parent_dir + '/' + name
+    try:
+        try:
```

Comment:
Please elaborate why you need two ```try``` blocks nested this tightly.

Reply:
Changed it.